src/engine/cql-to-elm.ts

```diff
@@ -1,72 +1,24 @@
 // External CQL -> ELM service
 import config from 'utils/ConfigManager';
-import {
-  extractJSONContent,
-  extractMultipartBoundary,
-  extractMultipartFileName
-} from 'utils/regexes';
-import { ElmLibrary, ElmLibraries } from 'elm-model';
+import { Client } from 'cql-translation-service-client';
+import { CqlLibraries as _cl } from 'cql-translation-service-client';
+import { ElmLibrary as BuilderElmLibrary, ElmLibraries as BuilderElmLibraries } from 'elm-model';
 
 const url = config.get('cqlToElmWebserviceUrl');
 
-export interface CqlLibraries {
-  [name: string]: {
-    cql: string;
-    version?: string;
-  };
-}
+const client = new Client(url);
 
 /**
  * Function that requests web_service to convert the cql into elm.
  * @param cqlLibraries - object containing CqlLibraries that is the input to the function.
  * @return The resulting elm translation of the cql libraries.
  */
-export function convertCQL(cqlLibraries: CqlLibraries): Promise<ElmLibraries> {
-  // Connect to web service
-  const formdata = new FormData();
-  Object.keys(cqlLibraries).forEach((key, i) => {
-    const cqlLibrary = cqlLibraries[key];
-    if (cqlLibrary.cql) {
-      formdata.append(`${key}`, cqlLibrary.cql);
-    }
-  });
-
-  return fetch(url, {
-    method: 'POST',
-    body: formdata
-  }).then(elm => {
-    const header = elm.headers.get('content-type');
-    let boundary = '';
-    if (header) {
-      // sample header= "multipart/form-data;boundary=Boundary_1"
-      const result = extractMultipartBoundary.exec(header);
-      boundary = result ? `--${result[1]}` : '';
-    }
-    const elmLibraries: ElmLibraries = {};
-    return elm.text().then(text => {
-      const elms = text.split(boundary).reduce((oldArray, line, i) => {
-        const body = extractJSONContent.exec(line);
-        if (body) {
-          const elmName = extractMultipartFileName.exec(line);
-          if (elmName) oldArray[elmName[1]] = JSON.parse(body[1]);
-        }
-        return oldArray;
-      }, elmLibraries);
-
-      return elms;
-    });
-  });
+export function convertCQL(cqlLibraries: CqlLibraries): Promise<BuilderElmLibraries> {
+  return client.convertCQL(cqlLibraries) as Promise<BuilderElmLibraries>;
 }
 
-export function convertBasicCQL(cql: string): Promise<ElmLibrary> {
-  // Connect to web service
-
-  return fetch(url, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/cql',
-      Accept: 'application/elm+json'
-    },
-    body: cql
-  }).then(elm => elm.json());
+export function convertBasicCQL(cql: string): Promise<BuilderElmLibrary> {
+  return client.convertBasicCQL(cql) as Promise<BuilderElmLibrary>;
 }
+
+export type CqlLibraries = _cl;
```

Comment:
Is there a reason we do this instead of just having other files `import { CqlLibraries } from 'cql-translation-service-client';`

Reply:
The main benefit to re-exporting here is that we don't need to change the other files and they can remain only dependent on this one file, and not this file and the exported type. Everything in this file is essentially just a wrapper around the cql-translation-service client methods, with updated type assertions, and types. This files real role now is the hide that dependency and do the necessary type assertions in one place.

I should note that the types provided by `cql-translation-service-client` are equivalent to the types in the pathway-builder, but still actually different so it needs the type assertion. There might be a better way to share and use the types between these two, but I didn't dig in too much into that and this seemed the least intrusive.

Reply:
Makes a lot of sense! LGTM - once the the client is updated you just need to change the dependencies right?

Reply:
Yup, waiting for https://github.com/mcode/cql-translation-service-client/pull/4 to get in and an updated package published to NPM.

I converted it to Draft so it's harder to accidentally merge 🙂 .